### PR TITLE
Activate + Retract

### PR DESCRIPTION
**Pull request by vladjdk**

### CODE REVIEW

File: src/components/free-willy/FreeWilly.tsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ export function FreeWilly() {
                 </Grid>
                 <Grid item xs={12}>
                     <Paper elevation={3}>
-                        <MyBids bethBids={bethBids} blunaBids={blunaBids}/>
+                        <MyBids bethBids={bethBids} blunaBids={blunaBids} onClaim={getUserBids}/>
```
Comment (theodubiel) [inline src/components/free-willy/FreeWilly.tsx]: change name of `onClaim` to `onBidUpdate`.  Unless this MyBids only does claiming.

File: src/components/free-willy/MyBids.tsx [hunk 2]
```diff
@@ -25,54 +26,160 @@ const columns = [
...
+        return () => {
+            clearInterval(intervalId);
+        }
+    }, [])
+    
+    
     useEffect(() => {
         const bids = [...bethBids, ...blunaBids];
         setRows(bids.map(bid => {
+            console.log(timestamp)
```
Comment (theodubiel) [inline src/components/free-willy/MyBids.tsx]: remove console.logs

File: src/components/free-willy/MyBids.tsx [hunk 3]
```diff
@@ -25,54 +26,160 @@ const columns = [
...
+                                if (row.bid_status != "Ready for activation") {
+                                    a = false;
+                                }
+                            });
+                            setActivatable(a)
+                        } else {
+                            setRetractable(false)
+                            setActivatable(false)
+                        }
+                        console.log(activatable)
```
Comment (theodubiel) [inline src/components/free-willy/MyBids.tsx]: remove console logs.  Also consider defining this function above out of the JSX logic.

File: src/hooks/useAnchorLiquidationContract.ts [hunk 4]
```diff
@@ -93,13 +88,22 @@ export const useAnchorLiquidationContract = (contractAddress: AccAddress) => {
...
-        const executeMsg = _createExecuteMsg(msg);
-      
-        return post({
-            msgs: [executeMsg],
-            fee: new Fee(fee.gas, { uusd: fee.amount }),
+        return _createExecuteMsg(msg);
+    }
+
+    function activateMultipleCollaterals(collaterals: string[]): MsgExecuteContract[] {
+        const msgs: MsgExecuteContract[] = []
```
Comment (theodubiel) [inline src/hooks/useAnchorLiquidationContract.ts:95]: Nice 👍 good to group these in one tx

File: src/components/free-willy/MyBids.tsx [hunk 5]
```diff
@@ -25,54 +26,160 @@ const columns = [
...
-            </Typography>
+            <Grid container alignItems="center">
+                <Grid item md={8} xs={6}>
+                    <Typography variant="h4" sx={{margin: '10px'}}>
+                            My Bids
+                    </Typography>
+                </Grid>
+                <Grid item md={4} xs={6} alignItems="right">
+                    <Button sx={{margin:"5px"}} variant="contained" disabled={!retractable} onClick={retract}>Retract Bid{selectionModel?.length!=1?"s":""}</Button>
+                    <Button sx={{margin:"5px"}} variant="contained" disabled={!activatable} onClick={activate}>Activate Bid{selectionModel?.length!=1?"s":""}</Button>
```
Comment (theodubiel) [inline src/components/free-willy/MyBids.tsx:157]: nice, I like the extra mile to add the `s`, most people would just do `Bid(s)`

File: src/hooks/useAnchorLiquidationContract.ts (hunk 4 above)
Comment (vladjdk) [reply src/hooks/useAnchorLiquidationContract.ts:95]: thanks!

**Pull request merged**